### DISCUSSION TITLE
HOWTO: Bundle server dependencies for compact distribution ready for docker

### DISCUSSION THREAD

**Discussion by cdaringe** [vercel/next.js#13916] [Help]
# Problem

- Next.js builds use `webpack` "external" for nodejs dependencies in my server bundles, but given that I am building from a monorepo, I do not want to include my full `node_modules`

# Discussion

Does anyone have a nice recipe for the webpack server build process to _bundle_ my nodejs dependencies into `.next/` such that the build output is truly _standalone_, and does not depend on any `node_modules` directories?

I see that some people use `pkg`, however, I'd like to be able to still a clear, clean node process, rather than a fancy binary. In this fashion, I still get good node ergonomics (optional sourcemaps, ability to enter a debug session remotely, etc).

**Comment by timche** [comment]
Dealing with this exact problem as well in a huge monorepo that is over 2GB and we don't want to copy the whole repo and all `node_modules` to the Docker image. 

It'd great if there could be an option to bundle the server as well, so we don't need `node_modules` at all. Best would be if the server could be simply started with `node .next/server.js` or similar, you get the idea.

CC @timneutkens

**Comment by jiricerhan** [comment]
Hi, I am working on the same thing, anyone has any progress on this?
